pg_shard.c

```diff
@@ -625,8 +634,13 @@ PlanSequentialScan(Query *query, int cursorOptions, ParamListInfo boundParams)
 	enable_indexscan = false;
 	enable_bitmapscan = false;
 
-	queryCopy = copyObject(query);
-	sequentialScanPlan = standard_planner(queryCopy, cursorOptions, boundParams);
+	/*
+	 * Remove all quals from query before calling planner. Remote workers will
+	 * apply quals for us: RowAndColumnFilterQuery ensures all quals are pushed
+	 * down during remote executions. The mutator returns a copy of the query.
+	 */
+	modifiedQuery = (Query *) RemoveFromExpressionQualsMutator((Node *) query, NULL);
```

Comment:
Why do we need the mutator construct here? Isn't setting the quals to NIL not sufficient?